05.ls/ls.rb

```diff
@@ -7,18 +7,21 @@
 SPACE_LENGTH = 3
 
 def main
-  files = search_files(ARGV[0])
+  files = search_files
   output(files)
 end
 
-def search_files(path)
+def search_files
+  option = OptionParser.new
+  params = {}
+  option.on('-a') { |v| params[:a] = v }
+  path = option.parse!(ARGV)[0]
   return [path] if FileTest.file?(path.to_s)
 
   dirpath = ARGV.empty? ? Dir.getwd : path
-  files = Dir.foreach(dirpath).reject do |file|
-    file.start_with?('.')
+  Dir.foreach(dirpath).sort.reject do |file|
+    file.start_with?('.') unless params[:a]
```

Comment:
unlessとstart_with?を重ねるとどういう条件で真/偽になるのかが直感的に把握しづらいので、A && BやA || Bの形式にした方が理解しやすい気がしますー。

```suggestion
  Dir.foreach(dirpath).sort.reject do |file|
    file.start_with?('.') && !params[:a]
```

Reply:
なるほど！このパターンでも&&、||が使えるんですね。修正しました。
今後も真偽条件を重ねるときはこの形式が使えないか検討するようにしていきます。